pkg/cloud/services/securitygroup/securitygroups.go

```diff
@@ -786,7 +785,51 @@ func ingressRulesFromSDKType(v *ec2.IpPermission) (res infrav1.IngressRules) {
 	return res
 }
 
-func (s *Service) getDefaultIngressRulesForControlPlaneLB() infrav1.IngressRules {
+// getIngressRulesToAllowKubeletToAccessTheControlPlaneLB returns ingress rules required in the control plane LB.
+// The control plane LB will be accessed by in-cluster components like the kubelet, that means allowing the NatGateway IPs
+// when using an internet-facing LB, or the VPC CIDR when using an internal LB.
+func (s *Service) getIngressRulesToAllowKubeletToAccessTheControlPlaneLB() infrav1.IngressRules {
+	if s.scope.ControlPlaneLoadBalancer() != nil && s.scope.ControlPlaneLoadBalancer().Scheme != nil && *s.scope.ControlPlaneLoadBalancer().Scheme == infrav1.ELBSchemeInternal {
```

Comment:
```suggestion
	if s.scope.ControlPlaneLoadBalancer() != nil && s.scope.ControlPlaneLoadBalancer().Scheme != nil && *s.scope.ControlPlaneLoadBalancer().Scheme == infrav1.ELBSchemeInternal {
```

So this happens often enough now, honestly that it's just annoying. :D 

If you don't mind, would you be so kind to add this function:

```go

func (e ELBScheme) Equal(other *ELBScheme) bool {
	if other == nil {
		return false
	}

	return e == *other
}
```

in api/v1beta2/network_types.go. Then you can just call it with:

```go
if s.scope.ControlPlaneLoadBalancer() != nil && infrav1.ELBSchemeInternal.Equal(s.scope.ControlPlaneLoadBalancer().Scheme) {
```

What do you think?

Reply:
Yeah, good idea! I'll do that.